src/t_stream.c

```diff
@@ -2025,10 +2026,12 @@ void xaddCommand(client *c) {
     }
 
     /* Append using the low level function and return the ID. */
+    errno = 0;
     streamID id;
     if (streamAppendItem(s,c->argv+field_pos,(c->argc-field_pos)/2,
         &id,parsed_args.id_given ? &parsed_args.id : NULL,parsed_args.seq_given) == C_ERR)
     {
+        serverAssert(errno != 0);
```

Comment:
so this is here to avoid future cases where someone will modify streamAppendItem to return C_ERR and will forget to set errno?
not sure how i feel about that.

Reply:
what's wrong with that?

Reply:
i guess nothing...

it feels a bit like (exaggerating) 
```c
int i = 1;
assert(i==1);
```

Reply:
except that, unlike your example, this assertion i added can actually fail